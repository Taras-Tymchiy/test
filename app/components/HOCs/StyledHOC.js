// @flow
import * as React from 'react';
import {
    Platform,
    StyleSheet
  } from 'react-native';
import { Post } from '../../entities/Post';

type StyledProps = {
  style?: any
}

export default function StyledHOC<Props: StyledProps, Component: React.ComponentType<Props>>(
    Comp: Component, defaultStyle: any
): React.ComponentType<Props>{
  return function Styled(props?: Props) {
    const mappedProps: Props = {
      ...props,
      style: [defaultStyle, props && props.style]
    };
    return (
      <Comp {...mappedProps} />
    );
  }
};
