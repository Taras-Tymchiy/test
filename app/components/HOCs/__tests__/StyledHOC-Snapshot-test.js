import React from 'react';
import renderer from 'react-test-renderer';
import StyledHOC from '../StyledHOC';
import { View } from 'react-native';

it('renders a StyledHOC wrapped View with empty styles using Snapshots', () => {
  const WrappedView = StyledHOC(View, {});
  expect(renderer.create(
    <WrappedView />
  )).toMatchSnapshot();
});
  
it('renders a StyledHOC wrapped View with styles using Snapshots', () => {
  const WrappedView = StyledHOC(View, { flex: 1});
  expect(renderer.create(
    <WrappedView />
  )).toMatchSnapshot();
});
  