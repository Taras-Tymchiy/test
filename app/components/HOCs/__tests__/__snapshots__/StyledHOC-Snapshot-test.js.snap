// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a StyledHOC wrapped View with empty styles using Snapshots 1`] = `
<View
  style={
    Array [
      Object {},
      undefined,
    ]
  }
/>
`;

exports[`renders a StyledHOC wrapped View with styles using Snapshots 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
